### PR TITLE
fix(aws-api-appsync): update getModelFields for flutter support

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:* https://github.com/aws-amplify/amplify-android/issues/1610

*Description of changes:*
- Update `getModelFields` method for flutter support to match the version used for native apps. The changes are identical to the changes in https://github.com/aws-amplify/amplify-android/pull/1585/files, but are applied to the version of `getModelFields` that is used for flutter support

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by poojamat — APPROVED** [review]

**Review by poojamat — COMMENTED** [review]
Would be nice to have a unit test for this.

**Comment by jamesonwilliams** [comment]
> Would be nice to have a unit test for this.

@jpignata doesn't agree

**Comment by Jordan-Nelson** [comment]
> Would be nice to have a unit test for this.

@poojamat I added a unit test for the new behavior for SerializedModel in aws-api-appsync. The unit tests seem to be failing in CI, although they are passing locally. I will look into that and update if needed.

File: aws-api-appsync/src/test/java/com/amplifyframework/api/aws/JustIDGraphQLRequestOptions.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (eeatonaws) [inline aws-api-appsync/src/test/java/com/amplifyframework/api/aws/JustIDGraphQLRequestOptions.java:2]: Should the copyright year be 2022?

Comment (Jordan-Nelson) [inline aws-api-appsync/src/test/java/com/amplifyframework/api/aws/JustIDGraphQLRequestOptions.java:2]:
```suggestion
 * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment (Jordan-Nelson) [reply aws-api-appsync/src/test/java/com/amplifyframework/api/aws/JustIDGraphQLRequestOptions.java]: Yes. Updated.

**Review by eeatonaws — APPROVED** [review]

**Pull request merged**